src/main/java/com/yahoo/bullet/kafka/KafkaSubscriber.java

```diff
@@ -40,67 +32,38 @@
      * @param maxUnackedMessages The maximum number of messages that can be received before an ack is needed.
      */
     public KafkaSubscriber(KafkaConsumer<String, byte[]> consumer, int maxUnackedMessages) {
+        super(maxUnackedMessages);
         this.consumer = consumer;
-        this.maxUnackedMessages = maxUnackedMessages;
-        this.receivedMessages = new LinkedList<>();
-        this.unackedMessages = new HashMap<>();
     }
 
-    /**
-     * Get the next batch of messages from Kafka and store it in the local buffer.
-     *
-     * @throws PubSubException when Kafka throws an error.
-     */
-    private boolean getMessages() throws PubSubException {
+    @Override
+    public List<PubSubMessage> getMessages() throws PubSubException {
         consumer.commitSync();
+        ConsumerRecords<String, byte[]> buffer;
         try {
             buffer = consumer.poll(0);
         } catch (KafkaException e) {
             throw new PubSubException("Consumer poll failed", e);
         }
+        List<PubSubMessage> messages = new ArrayList<>();
         for (ConsumerRecord<String, byte[]> record : buffer) {
             Object message = SerializerDeserializer.fromBytes(record.value());
             if (message == null || !(message instanceof PubSubMessage)) {
                 log.warn("Invalid message received: {}", message);
                 continue;
             }
-            receivedMessages.add((PubSubMessage) message);
+            messages.add((PubSubMessage) message);
         }
-        return !receivedMessages.isEmpty();
+        return messages;
     }
 
     @Override
     public PubSubMessage receive() throws PubSubException {
```

Comment:
Remove this method since it just calls super.receive().

Reply:
Will do. Forgot to remove after taking out the logs